sklearn/semi_supervised/tests/test_label_propagation.py

```diff
@@ -151,14 +152,16 @@ def test_convergence_warning():
     assert mdl.n_iter_ == mdl.max_iter
 
     mdl = label_propagation.LabelSpreading(kernel="rbf", max_iter=500)
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
```

Comment:
We prefer to explicitely pass the warning category that motivated this check. Based on the content of this test, it seems that it's a ConvergenceWarning